snuba/datasets/entities/events.py

```diff
@@ -161,6 +170,7 @@ def callback_func(
                             "query": format_query(query),
                             "primary_result": primary_result_data[idx],
                             "other_result": result_data[idx],
+                            "reason": reason,
```

Comment:
I don't know at what point this starts to impact performance, but it seems like this could be restructured so that you only have to loop through the results set once. Probably not a blocker right now but something to think about.

Reply:
Yeah, I'll try to restructure this in a future iteration. I'm not too concerned about optimising performance now since this callback is run in a thread after the request has already been returned to the client.